sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -176,4 +204,23 @@ public abstract partial class PipelineTransport : System.ClientModel.Primitives.
         protected abstract void ProcessCore(System.ClientModel.Primitives.PipelineMessage message);
         protected abstract System.Threading.Tasks.ValueTask ProcessCoreAsync(System.ClientModel.Primitives.PipelineMessage message);
     }
+    public partial class RequestOptions
+    {
+        public RequestOptions() { }
+        public System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
+        public System.ClientModel.Primitives.ClientErrorBehaviors ErrorOptions { get { throw null; } set { } }
+        public void AddHeader(string name, string value) { }
```

Comment:
TODO: Can we quickly add SetHeader to this PR, or would it be better to file an issue?

Reply:
The implementation is a bit non-trivial, tracking with https://github.com/Azure/azure-sdk-for-net/issues/41314